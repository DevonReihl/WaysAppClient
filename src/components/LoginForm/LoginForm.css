.LoginForm > div > label {
  font-weight: 600;
  font-size: 18px;
  color: #1b1b1b;
}

.LoginForm {
  padding: 10px;
  width: 100%;
  margin: auto;
}

.button-wrapper > a {
  color: rgb(46, 140, 216);
  font-weight: bold;
}

@media all and (min-width: 460px) {
  .LoginForm {
    width: 400px;
  }
}
