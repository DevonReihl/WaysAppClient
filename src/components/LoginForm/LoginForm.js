import React, { Component } from 'react';
import { Input, Label } from '../Form/Form';
import AuthApiService from '../../services/auth-api-service';
import UserContext from '../../contexts/UserContext';
import Button from '../Button/Button';
import './LoginForm.css';

class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {},
  };

  static contextType = UserContext;

  state = { error: null };

  firstInput = React.createRef();

  handleSubmit = (ev) => {
    ev.preventDefault();
    const { username, password } = ev.target;

    this.setState({ error: null });
    this.props.setLoading(true);
    AuthApiService.postLogin({
      username: username.value,
      password: password.value,
    })
      .then((res) => {
        username.value = '';
        password.value = '';
        this.context.processLogin(res.authToken);
        this.props.onLoginSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      })
      .finally(() => {
        this.props.setLoading(false);
      });
  };

  componentDidMount() {
    if (this.refs.firstInput) {
      this.firstInput.current.focus();
    }
  }

  render() {
    const { error } = this.state;
    return (
      <div className="form-wrapper">
        <form action="#" className="LoginForm" onSubmit={this.handleSubmit}>
          <div role="alert">{error && <p>{error}</p>}</div>
          <div>
            <Label htmlFor="login-username-input">Username</Label>
            <br />
            <Input
              ref={this.firstInput}
              id="login-username-input"
              name="username"
              required
            />
          </div>
          <div>
            <Label htmlFor="login-password-input">Password</Label>
            <br />
            <Input
              id="login-password-input"
              name="password"
              type="password"
              required
            />
          </div>
          <br />
          <div className="button-wrapper">
            <Button className="myButton" type="submit">
              Login
            </Button>
          </div>
        </form>
        <br />
        <div style={{ fontSize: '16px' }}>
          <center>
            <p>
              <b>Demo User:</b>
              <br />
              Username: John Rambo
              <br />
              Password: password{' '}
            </p>
          </center>
        </div>
      </div>
    );
  }
}

export default LoginForm;
