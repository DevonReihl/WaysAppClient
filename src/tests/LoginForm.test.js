import React from 'react'
import LoginForm from '../components/LoginForm/LoginForm'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`LoginForm component`, () => {
    it('Renders without crashing', () => {
        shallow(<LoginForm />)
    })
    it('Matches snapshot', () => {
        const wrapper =  shallow(
        <LoginForm />
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})