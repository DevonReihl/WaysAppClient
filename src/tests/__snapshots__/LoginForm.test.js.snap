// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component Matches snapshot 1`] = `
<div
  className="form-wrapper"
>
  <form
    action="#"
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div>
      <Label
        htmlFor="login-username-input"
      >
        Username
      </Label>
      <br />
      <ForwardRef
        id="login-username-input"
        name="username"
        required={true}
      />
    </div>
    <div>
      <Label
        htmlFor="login-password-input"
      >
        Password
      </Label>
      <br />
      <ForwardRef
        id="login-password-input"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <br />
    <div
      className="button-wrapper"
    >
      <ForwardRef
        className="myButton"
        type="submit"
      >
        Login
      </ForwardRef>
    </div>
  </form>
  <br />
  <div
    style={
      Object {
        "fontSize": "16px",
      }
    }
  >
    <center>
      <p>
        <b>
          Demo User:
        </b>
        <br />
        Username: John Rambo
        <br />
        Password: password
         
      </p>
    </center>
  </div>
</div>
`;
